src/core/resolver/xds/xds_dependency_manager.cc

```diff
@@ -349,17 +307,22 @@ void XdsDependencyManager::ResetBackoff() {
 }
 
 void XdsDependencyManager::OnListenerUpdate(
-    std::shared_ptr<const XdsListenerResource> listener) {
+    absl::StatusOr<std::shared_ptr<const XdsListenerResource>> listener) {
   GRPC_TRACE_LOG(xds_resolver, INFO)
       << "[XdsDependencyManager " << this << "] received Listener update";
   if (xds_client_ == nullptr) return;
+  if (!listener.ok()) {
+    current_listener_.reset();
+    return ReportError("LDS", listener_resource_name_,
+                       listener.status().message());
+  }
   const auto* hcm = absl::get_if<XdsListenerResource::HttpConnectionManager>(
-      &listener->listener);
+      &(*listener)->listener);
   if (hcm == nullptr) {
-    return OnError(listener_resource_name_,
-                   absl::UnavailableError("not an API listener"));
+    current_listener_.reset();
+    return ReportError("LDS", listener_resource_name_, "not an API listener");
   }
-  current_listener_ = std::move(listener);
+  current_listener_ = std::move(*listener);
```

Comment:
nit: I believe the recommendation is to have code as below - [internal link](https://g3doc.corp.google.com/devtools/library_club/g3doc/totw/181.md?cl=head#moving-from-the-value-of-an-abslstatusor):
```suggestion
  current_listener_ = *std::move(listener);
```